src/classes/CRLP_Rollup_SEL.cls

```diff
@@ -178,43 +444,65 @@ public class CRLP_Rollup_SEL {
      * @return List<DescribeFieldResult>
      */
     public static List<Schema.DescribeFieldResult> getReferencedFieldsByObject(SObjectType objType) {
-        String objName = UTIL_Describe.getObjectDescribe(objType).getName();
-        Map<String, Schema.DescribeFieldResult> allFields = UTIL_Describe.getAllFieldsDescribe(objName);
-        Map<String, Schema.DescribeFieldResult> referencedFields = new Map<String, Schema.DescribeFieldResult>();
-        for (Rollup__mdt rollup : cachedRollups) {
-            if (rollup.Active__c == true) {
-                if (rollup.Summary_Object__r.QualifiedApiName == objName) {
-                    String fldName = rollup.Summary_Field__r.QualifiedApiName;
-                    referencedFields.put(fldName, allFields.get(fldName.toLowerCase()));
-                }
-                if (rollup.Detail_Object__r.QualifiedApiName == objName && rollup.Detail_Field__c != null) {
-                    String fldName = rollup.Detail_Field__r.QualifiedApiName;
-                    referencedFields.put(fldName, allFields.get(fldName.toLowerCase()));
-                }
-                if (rollup.Amount_Object__r.QualifiedApiName == objName && rollup.Amount_Field__r != null) {
-                    String fldName = rollup.Amount_Field__r.QualifiedApiName;
-                    referencedFields.put(fldName, allFields.get(fldName.toLowerCase()));
-                }
-                if (rollup.Date_Object__r.QualifiedApiName == objName && rollup.Date_Field__r != null) {
-                    String fldName = rollup.Date_Field__r.QualifiedApiName;
-                    referencedFields.put(fldName, allFields.get(fldName.toLowerCase()));
-                }
-            }
+        // Collect SObjectFields in set in case there are duplicates
+        final Set<SObjectField> fields = new Set<SObjectField>();
+
+        // Always add the AccountId and Primary_Contact__c for the Opportunity object
+        if (objType == Schema.Opportunity.SObjectType) {
+            fields.add(Schema.Opportunity.AccountId);
+            fields.add(Schema.Opportunity.Primary_Contact__c);
         }
-        // Always include the CurrencyIsoCode field if it's available for the object
-        if (UserInfo.isMultiCurrencyOrganization() && allFields.containsKey('currencyisocode')) {
-            referencedFields.put('CurrencyIsoCode', allFields.get('currencyisocode'));
+
+        if (objType != null) {
+            final DescribeSObjectResult sObjectDescribe = UTIL_Describe.getObjectDescribe(objType);
+            final String sObjectName = sObjectDescribe.getName();
+            
+            // Note: DescribeSObjectResult.fields.getMap() is a case-insensitive map
+            final Map<String, SObjectField> fieldsByName = sObjectDescribe.fields.getMap();
+
+            // Always include the CurrencyIsoCode field if available for the object
+            if (
```

Comment:
In the case the `if` statement is on more than one line, I would vote for having a distinct line separating where the `if` ends and its block starts. However, having a variable clearly explaining what the expression is would be the best approach. For instance:
```
Boolean hasCurrencyIsoCode = UserInfo.isMultiCurrencyOrganization() && fieldsByName.get('CurrencyIsoCode') != null;
if (hasCurrencyIsoCode) {
    ...
}
```

Reply:
My personal preference is a rule that if any part inside a "bracket" (i.e. parentheses, curly bracket, or square bracket) needs to be on a separate line, all parameters/parts need to be on separate lines indented on more indent than the bracket.  (+ all end brackets should have the same indent as its start bracket).  **Purpose**: One doesn't have to search for commas which IMHO makes it easier to read.

```apex
// Not allowed
String message = String.format('format', new List<String> {
    'param1', 
    'param2'
});

// Allowed
String message = String.format(
    'format', 
    new List<String> {
        'param1', 
        'param2'
    }
);
```

@RadSFDeveloper , IMHO, your method doesn't make it easier to read and understand the Boolean.  The example only has a single AND condition which is fairly simple.  For more complex logic with nested ANDs and ORs, having all logic on one line makes it hard to read quickly and prone to making a parentheses mistake.


```apex
// CRLP_Rollup_SEL original line ; changed line 407:  setting includeRollupDefinition

// Original
includeRollupDefinition = ((r.Detail_Object__r.QualifiedApiName == oppObjectName || r.Detail_Object__r.QualifiedApiName == pmtObjectName)
	|| rollupType == CRLP_RollupProcessingOptions.RollupType.ContactHardCredit
        && r.Amount_Object__r.QualifiedApiName != pscObjectName && r.Amount_Object__r.QualifiedApiName != accSCObjectName);  // Hard to understand quickly?

// Proposed change
includeRollupDefinition = (
    (
        rollup.Detail_Object__r.QualifiedApiName == oppObjectName 
        || rollup.Detail_Object__r.QualifiedApiName == pmtObjectName
    )
    && rollup.Amount_Object__r.QualifiedApiName != pscObjectName 
    && rollup.Amount_Object__r.QualifiedApiName != accSCObjectName
); // Hard to understand quickly?  Is it an improvement?

Reply:
Thank you for the feedback @spelak-salesforce . I think there is a bit of misunderstanding, the logic should not be on one line: I agree that every `&&` and `II` *might* need a new line, so the second formatting of the `message` using `String.format()` is preferred. However, code style is sometimes an art, so it all depends.  The main objective is to encapsulate the boolean expression into a descriptive variable or a function and I hope the Clean Code rephrasing clarifies it: 
>Encapsulate Conditionals: Boolean logic is hard enough to understand without having to see it in the context of an if or while statement. Extract functions that explain the intent of the conditional.

For the example above, I would not have the expression in the main method but actually create a new function with descriptive name and extract the code into it so the main method is short and clear. If anyone reviewing/maintaining the class needs to read details, they can view the function definition. Meaning, the main method starts as the summary and the level of detail increases towards the end of class.
> The Newspaper Metaphor: Think of a well-written newspaper article. You read it vertically. At the top you expect a headline that will tell you what the story is about and allows you to decide whether it is something you want to read. ... We would like a source file to be like a newspaper article. The name should be sufficient to tell us whether we are in the right module or not. Detail should increase as we move downward until at the end we find the lowest level functions. A newspaper is composed of many articles: most are very small. Some are bit larger. Very few contain as much text as a page can hold. This makes the newspaper usable. If the newspaper were just one long story containing a disorganized agglomeration of facts, dates and names, then we simply would not read it.

So, I would extract the above expression into a separate method and hopefully give it a descriptive name:
```
private static Boolean shouldIncludeRollupHardCredit(String detailSObjectApiName, String amountSObjectApiName) {
    return 
        (detailSObjectApiName == OPPORTUNITY_SOBJECT || detailSObjectApiName == PAYMENT_SOBJECT)
        && amountSObjectApiName != PARTIAL_SOFT_CREDIT_SOBJECT 
        && amountSObjectApiName != ACCOUNT_SOFT_CREDIT_SOBJECT;
}
```
I will give more details on refactoring of this method in internal document.

Reply:
> the second formatting of the `message` using `String.format()` is preferred
I politely disagree that this is "better" because I, subjectively, find it difficult to chase down commas to delimit method parameters.  Though, I will and whole-heartedly agree to use the coding standard set forth.

The simple rule described below turns out to be a great way to standardize code format because it gets rid of the subjective nature of thinking something is easy to read.

Plus, the rule below can be applied among different Salesforce programming languages: Apex, CSS, HTML, JavaScript.

1. Define "bracket" to be are parentheses `()`, curly brackets `{}`, or square brackets `[]`
2. All closing brackets must have the same indentation as it's opening bracket
3. If any parameter/part contained in a bracket on a separate line than the opening bracket, it must have one (1) more indentation that the  bracket 
4. If there is more than 1 parameter/part contained in a bracket and at least one parameter/part is on a new line, all parameters must be on new lines